### ISSUE TITLE
hyperparams optimization

### ISSUE THREAD

**Issue by raulcarlomagno** [catboost/catboost#504]
catboost works in a decent way without hyperparams tuning
but im trying to get better models with bayesian hyperparameter optimization, i saw a few @annaveronika 's talks and im not sure about if i should tune learning_rate and iterations params, i think she says that is better to not optimize them?

right now, im optimizing:
-depth
-iterations
-border_count (i think this is not necessary right?)
-l2_leaf_reg
-learning_rate
-random_strength
-bagging_temperature (not tuned, im going to include this hyperparam in the tuning)

**Comment by annaveronika** [comment]
Here is a description of how to tune parameters:
https://tech.yandex.com/catboost/doc/dg/concepts/parameter-tuning-docpage/

You definitely should find decent number of iterations, this is the most important parameter.
You find a good pair learning_rate+iterations, you need to run the algorithm with default amount of iterations (1000). After that you need to look on the graph of the loss function on the evaluation dataset. You need the best iteration to be close to the last iteration. If it's too early you should decrease learning rate. It it's already too late (it's the very last iteration), then you see underfitting and you need to increase number of iterations or decrease learning rate.

You can use the plot=True parameter to observe the graphs.

**Comment by annaveronika** [comment]
about border_count - in some cases it can be important, we know such dataset. So you can try the default 128 and also try 254

**Comment by khrisanfov** [comment]
@annaveronika Hello! Do you suggest pick learning_rate+iterations first with default values of other parameters and then tune other parameters with picked combination of learning_rate+iterations?

**Comment by khrisanfov** [comment]
> If it's already too late (it's the very last iteration), then you see underfitting and you need to increase number of iterations or **decrease** learning rate.

I think it should be increased.

**Comment by annaveronika** [comment]
> > If it's already too late (it's the very last iteration), then you see underfitting and you need to increase number of iterations or **decrease** learning rate.
> 
> I think it should be increased.

Yes, you're right, thanks, corrected it.

**Comment by raulcarlomagno** [comment]
i think is gonna help me, i still have to try it, thank u!!!

**Comment by raulcarlomagno** [comment]
@annaveronika , sorry for writing again, but i think there is something that i dont understand. first of all, you recommend to get a good pair learning_rate+iterations, i dont understand the full procedure, sorry. and what about letting always catboost to set the learning rate automatically?
